pkg/util/clusterdata/cluster_version_feature.go

```diff
@@ -0,0 +1,55 @@
+package clusterdata
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+func newClusterVersionEnricherFeature(restConfig *rest.Config, oc *api.OpenShiftCluster) (enricherFeature, error) {
+	client, err := configclient.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &clusterVersionEnricherFeature{
+		client: client,
+		oc:     oc,
+	}, nil
+}
+
+type clusterVersionEnricherFeature struct {
+	client configclient.Interface
+	oc     *api.OpenShiftCluster
+}
+
+func (ef *clusterVersionEnricherFeature) Name() string {
+	return "cluster version"
+}
+
+func (ef *clusterVersionEnricherFeature) FetchData(ctx context.Context) (func(), error) {
```

Comment:
I think API subresources are a very long way away.  The advantage of not returning an error for now is less machinery to try and gather up errors from all the goroutines, especially when all we end up doing is to log the error anyway.  Logging the error locally also eases removing Name(), fwiw.

Passing oc into the callback: prevents a future developer from trying to access oc when they're not allowed to because they're not serialised.  Example: the following PR code is already pretty close to a data race:

```
func (ef *workerProfilesEnricherFeature) FetchData(ctx context.Context) (func(), error) {
	r, err := azure.ParseResourceID(ef.oc.ID)
```

Because you could already be reading ef.oc.ID at the same time as another goroutine is concurrently modifying oc in one of the apply() functions.

Reply:
For history: we decided to get rid of errors and log locally, but keep `oc` as a field of the struct & in constroctor to make it more clear that struct is bound to a specific object and to avoid the situations when a `restConfig` is created from one `oc` object, but the callback is called with a different one.